src/release/main.d

```diff
@@ -159,19 +159,58 @@ void main ( string[] params )
         if (readYesNoResponse("Would you like to close them now?"))
         {
             foreach (milestone; open_milestones)
-            {
-                writefln("Closing %s .. (%s open issues)",
-                         milestone.title, milestone.open_issues);
+                keepTrying(
+                {
+                    writef("Closing %s (%s open issues) ... ",
+                        milestone.title, milestone.open_issues);
 
-                updateMilestoneState(con, repo, milestone.number, State.closed);
-            }
+                    updateMilestoneState(con, repo, milestone.number, State.closed);
+                });
         }
     }
 
     writefln("All done.");
 }
 
 
+/*******************************************************************************
+
+    Repeatatly tries to call dg() if it threw an exception, while also keeping
+    the user informed about the attempts.
+
+    Works visually best if the dg() function prints it's actions according to
+    this pattern: "doing action abc ... " (no new line)
+
+    The function will then add either "success\n" or "failure: reason\n".
+
+    Params:
+        dg = delegate to call until it succeeds
+
+*******************************************************************************/
+
+void keepTrying ( void delegate ( ) dg )
+{
+    import core.thread;
+    import std.stdio;
+
+    while (true) try
+    {
+        dg();
+
+        stdout.flush();
+        break;
+    }
+    catch (Exception exc)
+    {
+        writefln("failed: %s", exc.msg);
+        writefln("retrying ...");
+        Thread.sleep(1.seconds);
```

Comment:
Also a ["human" exponential backoff](https://github.com/sociomantic-tsunami/git-hub/blame/2b1f5329c41480ccd9599779faff23bda736e163/git-hub#L863-L876) could be a good idea :P